scripts/india_rbi/below_poverty_line/preprocess.py

```diff
@@ -0,0 +1,164 @@
+INDIA_ISO_CODES["Lakshwadeep"] = "IN-LD"
+INDIA_ISO_CODES["Pondicherry"] = "IN-PY"
+INDIA_ISO_CODES["Puducherry"] = "IN-PY"
+INDIA_ISO_CODES["Dadra and Nagar Haveli and Daman and Diu"] = "IN-DN_DD"
+INDIA_ISO_CODES["Telangana"] = "IN-TG"
+
+
+class BelowPovertyLineDataLoader:
+    COLUMN_HEADERS = [
+        "year", "territory", "count_person_rural", "percentage_person_rural",
+        "poverty_line_rural", "count_person_urban", "percentage_person_urban",
+        "poverty_line_urban", "count_person_combined",
+        "percentage_person_combined"
+    ]
+
+    def __init__(self, source):
+        self.source = source
+        self.raw_df = None
+        self.clean_df = None
+
+    def download(self):
+        df = pd.read_excel(self.source)
+        #Drop title rows and empty rows
+        df = df.iloc[2:40]
+        #Remove emprty columns
+        df = df.drop(df.columns[0], axis=1)
+        self.raw_df = df.drop(df.columns[19], axis=1)
+
+    def _setup_location(self):
+        self.clean_df["territory"] = self.clean_df["territory"].apply(
+            lambda x: INDIA_ISO_CODES[x])
+
+    def _make_column_numerical(self, column):
+        self.clean_df[column] = self.clean_df[column].astype(str).str.replace(
+            ',', '')
+        self.clean_df[column].loc[self.clean_df[column] == '.'] = np.nan
+        self.clean_df[column] = pd.to_numeric(self.clean_df[column])
+
+    def process(self):
+        #Get yearly date and set the date
+        df2004_05 = self._process_yearly_dataframe(year="2005-03",
+                                                   start_index=0)
+        df2009_10 = self._process_yearly_dataframe(year="2010-03",
+                                                   start_index=10)
+        df2011_12 = self._process_yearly_dataframe(year="2012-03",
+                                                   start_index=20)
+        combined_df = df2004_05.append(df2009_10)
+        self.clean_df = combined_df.append(df2011_12)
+
+        self._make_column_numerical("count_person_rural")
+        self._make_column_numerical("percentage_person_rural")
+        self._make_column_numerical("poverty_line_rural")
+        self._make_column_numerical("count_person_urban")
+        self._make_column_numerical("percentage_person_urban")
+        self._make_column_numerical("poverty_line_urban")
+        self._make_column_numerical("count_person_combined")
+        self._make_column_numerical("percentage_person_combined")
+
+        #Count columns are in thousands, hence multiply them by 1000
+        self.clean_df[
+            "count_person_rural"] = self.clean_df["count_person_rural"] * 1000
+        self.clean_df[
+            "count_person_urban"] = self.clean_df["count_person_urban"] * 1000
+        self.clean_df["count_person_combined"] = self.clean_df[
+            "count_person_combined"] * 1000
+
+        self.clean_df.drop(
+            self.clean_df.index[self.clean_df.territory == "All India"],
```

Comment:
Should we include the all India category too?

Reply:
Done.